hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/properties/SerializableSemVers.java

```diff
@@ -99,6 +100,17 @@ public boolean hasMigrationRecordsFrom(@Nullable final SoftwareVersion other) {
         return isNonPatchUpgradeFrom(other) || (this.isAfter(other) && currentVersionHasPatchMigrationRecords);
     }
 
+    public boolean isNonConfigUpgrade(@Nullable final SoftwareVersion other) {
+        if (other == null) {
```

Comment:
We can prob drop this check now and `requireNonNull(other)`. (I think it would mean a catastrophic failure if somehow a saved state was loaded with a null `SoftwareVersion`.)

Reply:
got it ! Changed in all places where we are adding `@Nullable final SoftwareVersion other` in this class

Reply:
Seems `null` is being used as `deserializedVersion` on `genesisInit`. Reverted back and added comment.